pkg/sharky/shard.go

```diff
@@ -114,6 +115,7 @@ LOOP:
 			for {
 				select {
 				case sh.errc <- sh.read(op):
```

Comment:
@janos I am puzzled by how this specific syntax usage would unfold. So I assume that if there is a reader on `errc` on the other side the specific case will be selected, blocking the possible selects on the other cases?
Or does `sh.read()` have to first return for the case to actually be selected?
I am concerned that `sh.read` might execute halfway while the context expires, resulting in the shard again not being able to drain the channel since nothing is written to it.

Reply:
ok question answered here: https://go.dev/play/p/-8VDC_2ACD7
it is enough to just have the reader on the channel, and the case will already be selected

Reply:
In which case it might make sense to even get rid of the context altogether and just not respect it when we get to the actual read part (we could respect it until the task is handed over to the shard, but not after).
Would be happy for some thoughts on this.

Reply:
That is right. The read function will not block the select case, as the channel will be written to only after the function has returned.

Offloading cancelation to the Read caller, but I think that currently, the already existing channels handling is convenient to be used for particular reads cancelation as well. As file operations could not be canceled with context, there is always a possibility to have some goroutines blocking.

Reply:
I see context here more like an insurance that cancelation will happen regardless if some code further away is changed. Code always can be more readable and improvements in that way here may result in better understanding of the flow. Somehow, I tend to keep operations on a single channel as close in the code as possible, preferably within one function with clear read only channels returned, if needed. If channels are more sparse in the code, it is a bit harder to follow.

Reply:
:+1: I added a small comment about the change in `store.go` just so that there's a breadcrumb trail in the code, and I shall keep the change as-is. Thanks for the feedback